server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -1267,3 +1268,117 @@ def _add_feature_stats(
+
+        if not has_mm and any(keys_not_none):
+            # An unexpected situation in which only some of the mandatory MM secrets are set
+            set_secrets = [
+                secret
+                for (not_none, secret) in zip(keys_not_none, mandatory_secrets)
+                if not_none
+            ]
+            logger.warn(
+                "Not all of the mandatory model monitoring secrets are set in the project's secrets. "
+                "Assuming the project has no model monitoring in place.",
+                project_name=self._project,
+                mandatory_secrets=mandatory_secrets,
+                set_secrets=set_secrets,
+            )
+
+        return has_mm
+
+    def _get_profile(
+        self, get_profile_function: Callable
+    ) -> Optional[mlrun.datastore.datastore_profile.DatastoreProfile]:
+        if not self._has_mm:
+            return
+        try:
+            return get_profile_function(
+                project=self._project, secret_provider=self._secret_provider
+            )
+        except mlrun.errors.MLRunNotFoundError as err:
+            # An unexpected situation in which the secrets are set but the datastore profile was removed
+            logger.warn(
+                "The project is marked as having model monitoring according to the secrets, "
+                "but the profile was not found.",
+                project_name=self._project,
+                err=mlrun.errors.err_to_str(err),
+            )
+
+    def _get_stream_profile(
+        self,
+    ) -> Optional[mlrun.datastore.datastore_profile.DatastoreProfile]:
+        return self._get_profile(
+            get_profile_function=mlrun.model_monitoring.helpers._get_stream_profile
+        )
+
+    def _get_tsdb_profile(
+        self,
+    ) -> Optional[mlrun.datastore.datastore_profile.DatastoreProfile]:
+        return self._get_profile(
+            get_profile_function=mlrun.model_monitoring.helpers._get_tsdb_profile
+        )
+
+    def delete(self) -> None:
+        if not self._stream_profile or not self._tsdb_profile:
+            logger.debug(
+                "No model monitoring for the project",
+                project_name=self._project,
+            )
+            return
+        try:
+            # Delete model monitoring resources
+            ModelEndpoints.delete_model_endpoint_monitoring_resources(
+                project_name=self._project,
+                db_session=self._db_session,
+                tsdb_profile=self._tsdb_profile,
+                stream_profile=self._stream_profile,
+                model_monitoring_applications=self._model_monitoring_applications,
+                model_monitoring_access_key=self._model_monitoring_access_key,
+            )
+        except Exception as exc:
+            logger.warning(
+                "Failed to delete model monitoring resources",
+                project_name=self._project,
+            )
+            raise exc
```

Comment:
Are you sure that we don't break the delete project API if for some reason the mm resources were not deleted  as expected in this case? these resources can be stored on the user's side so what happens if the user has changed his kafka creds for example? we should catch the the exception and print it out but skip the rmm esources deletion in this case.

Reply:
I agree it's a tricky point, but I kept the behavior here as it was.
I think it's better to start more strict and loosen this approach if we see fit.